fs/move.py

```diff
@@ -64,6 +64,9 @@ def move_file(
     with manage_fs(src_fs, writeable=True) as _src_fs:
         with manage_fs(dst_fs, writeable=True, create=True) as _dst_fs:
             if _src_fs is _dst_fs:
+                if src_path == dst_path:
+                    return
```

Comment:
Does this need to do the `normpath` thing first, in case one of the paths has `../` etc.?
Also, this might need to do a case-insensitive comparison on case-insensitive filesystems?

Reply:
Hmm, it's outside the scope of **this** PR, but I wonder if e.g. a `are_same_path(path1, path2)` would be a useful addition to https://docs.pyfilesystem.org/en/latest/interface.html (which would handle the normalising and case-insensitivity-ness for you automatically) ? :thinking:

Reply:
I don't think this is enough. We'd need a function to check if two tuples `(fs, path)` point to the same resource (I had this problem before and started working on it in #519)

Reply:
Nope, I was only suggesting comparing paths (within a filesystem), NOT comparing entire filesystems, or looking at the actual resources themselves. The only "link" to the filesystem is so that it can call the `validatepath` function on each of the paths and do a case-insensitive comparison if the filesystem is itself case-insensitive. Semantically, it's only an enhanced version of `==` :slightly_smiling_face:

Reply:
Yes that would be a good first step - I have not tested this yet, but let's say you move a file from one connection to a ftp server onto itself in another connection to the same ftp server. This would just delete the file, wouldn't it?

This is certainly out of scope for this PR, shall I open a new issue for this?

Reply:
I updated this to compare the normpathes. If you have a case-insensitive filesystem this would not exit early but delegate the move to the given FS, which should work normally but at this time triggers the delete on move bug. Exiting a little bit earlier is nice, but - as you said - a `are_same_path` function would be right approach here.

Reply:
Created #545 for tracking this.